src/mammos_mumag/hysteresis.py

```diff
@@ -58,6 +61,9 @@ def run(
     if hstep is None:
         hstep = (hfinal - hstart) / hnsteps
 
+    if not isinstance(mesh, Mesh):
+        mesh = Mesh(mesh)
```

Comment:
I think we should maybe skip this and leave it to Simulation to convert to the required type.